apiserver/src/api/mod.rs

```diff
@@ -232,6 +243,29 @@ pub async fn run_server<T: 'static + BottlerocketShadowClient>(
     Ok(())
 }
 
+// The certificate is refreshed periodically (default 60 days). Once the certificate is renewed, the apiserver
+// needs to stop in order to reload the new certificate.
+// We cache the certificate initially when brupop starts the server, and compare it to the update-to-date certificate periodically.
+// If they don't match, we recognize it as a new certificate, so the server needs to be restarted.
+async fn reload_certificate(
+    server: Server,
```

Comment:
Sorry to comment on an old PR, but I was catching up on some of the stuff that's changed in brupop!

I wonder if it makes more sense to make this a mutable reference? It seems unusual to clone it. Although, I suppose if it works, Actix must have done some work to make sure that the interior state of the Server is thread-safe.

Reply:
I actually find out that using https://github.com/notify-rs/notify to watch the fs change and reload the certificate is what other people like to do.

Reply:
@cbgbt - This has actually changed recently: https://github.com/bottlerocket-os/bottlerocket-update-operator/commit/561e6c0c1251cee2c9c6694232fae52564ce6419#diff-5cad36d03cb8802a9c5a0a20ae2e841d683e1cd564146c441e2940fb1ecf652eR255

A few APIs changed in Actix so cloning the server and using it directly wasn't a great option. Instead, now, we can use a `actix_web::dev::ServerHandle` as a "server handler" to call into it and stop it like:
```rust
 server_handler.stop(true).await;
```